src/device/array.jl

```diff
@@ -1,6 +1,23 @@
 # Contiguous on-device arrays
 
-export ROCDeviceArray, ROCDeviceVector, ROCDeviceMatrix, ROCBoundsError
+export ROCDevice, ROCDeviceArray, ROCDeviceVector, ROCDeviceMatrix, ROCBoundsError
+
+
+# ToDo: This is not sufficient/general enough (doesn't support multiple devices)
+# What's the AMDGPU equivalent of CUDA.CuDevice?
+"""
+    struct ROCDevice <: AbstractGPUDevice
+
+A ROC device.
+"""
+struct ROCDevice <: AbstractGPUDevice end
```

Comment:
Can you instead change https://github.com/JuliaGPU/AMDGPU.jl/blob/99967b75deb4d78c12236e9a77943b4e0294af35/src/agent.jl#L3 to `mutable struct HSAAgent <: AbstractGPUDevice`? We use `HSAAgent` for most operations involving devices.

Reply:
Thanks, done! How do I do

```julia
Adapt.adapt_storage(device::HSAAgent, x) = adapt_storage(ROCArray, x)
```

correctly (i.e. send the array to the right device)?